### PR TITLE
WIP: process config through template engine

### PR DESCRIPTION
**Pull request by stambata**
WIP!!! Don't merge.
TO DO: add documentation.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,10 +1,27 @@
 /* eslint no-process-env:0 */
 const rc = require('rc');
 const merge = require('ut-function.merge');
+const template = require('ut-function.template');
 const path = require('path');
 const fs = require('fs');
 const serverRequire = require;
 
+function interpolate(what = {}, context = {}) {
```
Comment (kalinkrustev) [inline index.js]: We should move this to ut-function.template.
If we pass object, then it will execute this code.

File: index.js [hunk 2]
```diff
@@ -108,7 +125,7 @@ function load({ params, app, method, env, root, version, resolve, config } = {})
 
     configs.unshift(baseConfig);
 
-    return merge(configs, mergeOptions);
+    return interpolate(merge(configs, mergeOptions), context);
```
Comment (kalinkrustev) [inline index.js]: Call template only if `typeof context === 'object'`.

**Review by kalinkrustev — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (stambata) [reply index.js]: done! https://github.com/softwaregroup-bg/ut-function/pull/11

**Comment by stambata** [comment]
Depends on: https://github.com/softwaregroup-bg/ut-function/pull/11

**Review by kalinkrustev — APPROVED** [review]

**Pull request merged**